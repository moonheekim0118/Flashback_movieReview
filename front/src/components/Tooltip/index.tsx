import React from 'react';
import styled from 'styled-components';

// buttons types
interface buttons {
  title: string;
  onClick: (e: any) => void;
}

interface Props {
  onClose: (e: any) => void;
  buttonList: Array<buttons>;
}

// 툴팁
const Tooltip = ({ onClose, buttonList }: Props) => {
  return (
    <>
      <Overaly onClick={onClose} className="overaly" />
      <Container>
        {buttonList.map((v, i) => (
          <ItemContainer key={v.title + i} className="item" onClick={v.onClick}>
            <Item>{v.title}</Item>
          </ItemContainer>
        ))}
      </Container>
    </>
  );
};

const Overaly = styled.div`
  position: fixed;
  top: 0;
  left: 0;
  right: 0;
  bottom: 0;
  z-index: 500;
`;

const Container = styled.div`
  display: flex;
  flex-direction: column;
  position: absolute;
  right: 30px;
  top: 15px;
  width: 180px;

  font-size: 1.2rem;
  background-color: #cc00cc;
  box-shadow: 0px 0px 11px -1px rgba(0, 0, 0, 0.75);

  z-index: 1000;
`;

const ItemContainer = styled.div`
  cursor: pointer;
  transition: 0.2s background-color ease-in-out;

  &:hover {
    background-color: #e6b3cc;
  }
`;

const Item = styled.div`
  padding: 10px 20px;
`;

export default Tooltip;
