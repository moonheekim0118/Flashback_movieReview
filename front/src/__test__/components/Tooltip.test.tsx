import 'jsdom-global/register'; 
import React from "react";
import { mount } from 'enzyme';
import Tooltip from '../../components/Tooltip';

describe('<Tooltip/>', () => {
   let container = null;
   const mockFn1 = jest.fn();
   const mockFn2 = jest.fn();
   const mockFn3 = jest.fn();

   const mockProps = [ { title: 'test1', onClick:mockFn1},
                       { title: 'test2', onClick:mockFn2} ];

   it('renders correctly', ()=>{ // 렌더링 테스트 
    container=mount(<Tooltip onClose={mockFn3} buttonList={mockProps}/>);
   });

   it('matches snapshot', ()=>{ // 스냅샷 비교 
    expect(container.html()).toMatchSnapshot();
   });

   it('should have two items', ()=>{ // item이 mockProps 개수대로 들어가있는지 확인 
      expect(container.find('#item').hostNodes()).toHaveLength(mockProps.length);
   });

   it('should call onClose if Overaly is clicked',()=>{ // onClose 버튼 시뮬레이션 
    container.find('#overaly').hostNodes().simulate('click');
    expect(mockFn3).toHaveBeenCalled();
    });

   it('should call first onClick if first Item container is clicked', ()=>{ // 첫번째 onClick 버튼 
    container.find('#item').first().simulate('click');
    expect(mockFn1).toHaveBeenCalled();
   });

   it('should call second onClick if second Item container is clicked', ()=>{ // 두번째 onClick 버튼 
    container.find('#item').last().simulate('click');
    expect(mockFn2).toHaveBeenCalled();
   });

})
