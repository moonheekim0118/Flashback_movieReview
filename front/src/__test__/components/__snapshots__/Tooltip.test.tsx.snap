// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Tooltip/> matches snapshot 1`] = `"<div id=\\"overaly\\" class=\\"Tooltip__Overaly-sc-1s0dps1-0 gUipaz\\"></div><div class=\\"Tooltip__Container-sc-1s0dps1-1 dUDDKK\\"><div id=\\"item\\" class=\\"Tooltip__ItemContainer-sc-1s0dps1-2 lmXqez\\"><div class=\\"Tooltip__Item-sc-1s0dps1-3 fTYYPM\\">test1</div></div><div id=\\"item\\" class=\\"Tooltip__ItemContainer-sc-1s0dps1-2 lmXqez\\"><div class=\\"Tooltip__Item-sc-1s0dps1-3 fTYYPM\\">test2</div></div></div>"`;
